### PR TITLE
TIP-405: add spin to check row

### PR DESCRIPTION
**Pull request by juliensnz**

### CODE REVIEW

File: features/Context/DataGridContext.php [hunk 1]
```diff
@@ -668,14 +668,18 @@ public function iCheckTheRows($rows)
...
-            $checkbox = $gridRow->find('css', 'td.boolean-cell input[type="checkbox"]:not(:disabled)');
+            $this->spin(function () use ($row) {
+                $gridRow  = $this->datagrid->getRow($row);
+                $checkbox = $gridRow->find('css', 'td.boolean-cell input[type="checkbox"]:not(:disabled)');
 
-            if (!$checkbox) {
-                throw $this->createExpectationException(sprintf('Unable to find a checkbox for row %s', $row));
-            }
+                if (null !== $checkbox) {
+                    $checkbox->check();
```
Comment (jmleroux) [inline features/Context/DataGridContext.php]: Does the check action really need to be in the spin ?
Spinning should be only for finding elements.

**Comment by jmleroux** [comment]
GTM after answer

File: features/Context/DataGridContext.php (hunk 1 above)
Comment (juliensnz) [inline features/Context/DataGridContext.php]: nope, not always. As the checkbox can disappear :/

**Comment by phaseinducer** [comment]
GTM

**Pull request merged**